inventory/s3/orc_reader.go

```diff
@@ -0,0 +1,155 @@
+package s3
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/go-openapi/swag"
+	"github.com/hashicorp/go-multierror"
+	"github.com/scritchley/orc"
+	"modernc.org/mathutil"
+)
+
+type OrcInventoryFileReader struct {
+	reader                   *orc.Reader
+	c                        *orc.Cursor
+	ctx                      context.Context
+	selectIndexByField       map[string]int
+	actualColumnIndexByField map[string]int
+	mgr                      *Reader
+	key                      string
+}
+
+type orcFile struct {
+	key           string
+	localFilename string
+	ready         bool
+}
+
+func getSelectFields(typeDescription *orc.TypeDescription) ([]string, map[string]int, map[string]int) {
```

Comment:
Please document this function, and consider also naming the return elements.  It has multiple return values of complex type, two of which have the same type.  I cannot understand the intent without reading implementation and usage.